### PR TITLE
NFDIV-2209 - Add CitizenResendInvite event

### PR DESCRIPTION
**Pull request by danielbe11**
### Change description ###

- Add CitizenResendInvite event to allow applicant 1 to update applicant 2's email address and send another case invite on FE.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-2209

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java [hunk 1]
```diff
@@ -156,6 +159,16 @@ public static List<String> validateCaseFieldsForIssueApplication(MarriageDetails
...
     }
 
+    public static List<String> validateCitizenResendInvite(CaseDetails<CaseData, State> details) {
+        var data = details.getData();
+        boolean isApplicant2EmailUpdatePossible = details.getState() == State.AwaitingApplicant2Response
+            && data.getCaseInvite().accessCode() != null
+            && data.getApplicationType() == ApplicationType.JOINT_APPLICATION;
+        return isApplicant2EmailUpdatePossible
+            ? emptyList()
+            : singletonList("Not possible to update email address");
```
Comment (lara-mpt) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java:169]: could be clearer -  "Not possible to update applicant 2 invite email address"

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenResendInvite.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            if (!validationErrors.isEmpty()) {
+                log.info("Validation errors: {} ", validationErrors);
+
+                return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                    .data(data)
+                    .errors(validationErrors)
+                    .build();
+            }
+            log.info("Setting new due date for Case Id: {} (citizen-resend-invite)", details.getId());
+            data.setDueDate(LocalDate.now().plus(2, ChronoUnit.WEEKS));
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenResendInvite.java:63]: should be stored in a constant which should signify why 2 weeks

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java [hunk 3]
```diff
@@ -156,6 +159,16 @@ public static List<String> validateCaseFieldsForIssueApplication(MarriageDetails
...
     }
 
+    public static List<String> validateCitizenResendInvite(CaseDetails<CaseData, State> details) {
+        var data = details.getData();
+        boolean isApplicant2EmailUpdatePossible = details.getState() == State.AwaitingApplicant2Response
+            && data.getCaseInvite().accessCode() != null
+            && data.getApplicationType() == ApplicationType.JOINT_APPLICATION;
+        return isApplicant2EmailUpdatePossible
+            ? emptyList()
+            : singletonList("Not possible to update applicant 2 invite email address");
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java:169]: Will this message be shown on the UI if yes worth checking with BA's

Comment (danielbe11) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java:169]: The event will only be for citizens so the error message will just be read in logs. Is that ok?

File: src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenResendInvite.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+
+        configBuilder
+            .event(CITIZEN_RESEND_INVITE)
+            .forStates(AwaitingApplicant2Response)
+            .name("Update applicant 2 email")
+            .description("Citizen event for applicant 1 to update applicant 2 email pre-submission")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CREATOR);
```
Comment (danielbe11) [inline src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenResendInvite.java:45]: I only want this event for citizen applicant 1 - is this the correct grant permission? @nitinprabhu

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/citizen/event/CitizenResendInvite.java:45]: yes that's correct @danielbe11

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java (hunk 3 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/validation/ValidationUtil.java:169]: ok if it is in only in logs that should be fine thanks

**Review by nitinprabhu — APPROVED** [review]

**Pull request closed without merging**